### PR TITLE
Binary expressions

### LINKED ISSUE #57: Opaque SyntaxElement - Binary Expression
### Details
Update the way we parse binary expressions to the new opaque `SyntaxElement` api. There is existing, commented out code that you can start with, but you will need to verify correct parsing according to PEMDAS. 
For example: `2 + (3 * 4) ^ 2 - 5 / 2 ^ (1 + 2)`
```
                           `-`
                         /      \
                       `+`      `÷`
                       /  \      / \
                      2   `^`   5  `^`
                          / \     /   \
                       `*`   2   2    `+`
                       / \            / \
                      3  4           1   2
```

### Issue Type
- [X] Feature
- [ ] Bugfix
- [ ] Documentation
- [X] Refactoring
- [ ] Other:

### Requirements
- Correct Implementation of PEMDAS
- Integration with the new `SyntaxElement` API

### Notes

### PR DESCRIPTION
**Pull request by Dalyj24**
This closes #57.

### CODE REVIEW

File: frontend/src/parser/parser_core.rs
Comment (CalebLItalien) [inline frontend/src/parser/parser_core.rs:1]: Let's move `self.current -= 1` into a standalone function off of the `Parser` struct, and invoke it at the beginning of `parse_binary_expression()`.

File: frontend/src/parser/binop_precedence.rs
Comment (CalebLItalien) [inline frontend/src/parser/binop_precedence.rs:1]: Can we move `parse_binary_expression` into this file, rename the file to `parse_binary_exp`, and make all these functions private?

File: frontend/src/parser/mod.rs
Comment (CalebLItalien) [inline frontend/src/parser/mod.rs:1]: Make this a private module again after making the above change.

File: frontend/src/parser/parse_expression.rs [hunk 1]
```diff
@@ -85,43 +91,122 @@ impl Parser {
         Ok(Some(assignment))
         
     }
-    
-    /// Parses a binary expression
+
+    /// Function for the parsing of binary expressions.
```
Comment (CalebLItalien) [inline frontend/src/parser/parse_expression.rs:95]: Very elegant solution!!!! :)

**Review by CalebLItalien — COMMENTED** [review]
Looks good! A few things of note.

**Review by kjwelch — APPROVED** [review]
Looks good to me

**Pull request merged**